src/discord.ts

```diff
@@ -0,0 +1,83 @@
+import { Client, GuildMember } from "discord.js";
+import { onSelector } from "./store";
+import { createSelector } from "@reduxjs/toolkit";
+import { getLinkedAccounts } from "./store/link";
+import { getParticipants, getHasSeenStart, getCallIsActive } from "./store/zoom";
+import { DISCORD_ACTIVE_ROLE } from "./config";
+
+
+const getActiveDiscordUsers = createSelector(
+	getLinkedAccounts, 
+	getParticipants, 
+	(linkedAccounts, participants) => {
+		return participants.map(({ id }) => linkedAccounts.byId[id].discord)
+});
+
+const getZoomInfo = createSelector(
+  getCallIsActive,
+  getHasSeenStart,
+  getParticipants,
+  (active, hasSeenStart, participants) => ({
+    active,
+    hasSeenStart,
+    participants,
+  })
+)
+
+// keeps discord in sync with the server side state
+export const syncDiscordStatus = (discord: Client) => {
+  // get the active role on each server
+  discord.guilds.cache.forEach(async (guild) => {
+    console.log(`Syncing to ${guild.name}`);
+    const roles = await guild.roles.fetch();
+    // role to apply/remove
+    const activeRole = roles.cache.find(role => role.name === DISCORD_ACTIVE_ROLE);
+    if (!activeRole) {
+      throw `${guild.name} does not have a ${DISCORD_ACTIVE_ROLE} role.`;
+    }
+
+    
+    // keep roles in sync
+    onSelector(getActiveDiscordUsers, async (activeUsers) => {
+      // ensure members are up to date
+      await guild.members.fetch();
+      // update roles
+      // filter is to prevent spamming the API with meaningless requests
+      const shouldUpdateMember = (member: GuildMember): boolean => {
+        return activeUsers.indexOf(member.id) > -1 || Boolean(member.roles.cache.find(role => role.id === activeRole.id));
```

Comment:
Not sure what you mean - `Boolean()` is an explicit cast, `new Boolean()` is the one which creates non-primitive values.

Reply:
Huh... I didn't know the difference between `Boolean()` and `new Boolean()`, I thought they were both go-home-Javascript-you're-drunk types of things. I stand by that `!!whatever` is better to avoid this confusion, but I'm a lot less concerned about it now.

Reply:
Agreed. I switched to !! because I was completely unaware of `new Boolean` and it seems like unnecessary confusion